django/cantusdb_project/main_app/views/chant.py

```diff
@@ -1619,18 +1638,37 @@ def get_chants_with_folios(chants_in_feast):
             current_chant = Chant.objects.filter(pk=pk).first()
             cantus_id = current_chant.cantus_id
 
-            request = requests.get(
-                f"https://cantusindex.org/json-cid/{cantus_id}",
-                timeout=5,
-            )
-            request_text = json.loads(request.text[2:])
-            if request_text:
-                context["suggested_fulltext"] = request_text[0]["fulltext"]
+            try:
```

Comment:
I see this try/except code repeated a few times, I wonder if it would be possible/worth putting this in a separate function